packages/block-library/src/image/view.js

```diff
@@ -54,19 +55,19 @@ const { state, actions, callbacks } = store(
 				const ctx = getContext();
 
 				// Bails out if the image has not loaded yet.
-				if ( ! ctx.imageRef?.complete ) {
+				if ( ! state.metadata[ ctx.imageId ].imageRef?.complete ) {
 					return;
 				}
 
-				// Stores the positons of the scroll to fix it until the overlay is
+				state.overlayEnabled = true;
+
+				// Stores the positions of the scroll to fix it until the overlay is
 				// closed.
 				state.scrollTopReset = document.documentElement.scrollTop;
 				state.scrollLeftReset = document.documentElement.scrollLeft;
 
-				// Moves the information of the expaned image to the state.
-				ctx.currentSrc = ctx.imageRef.currentSrc;
-				state.currentImage = ctx;
-				state.overlayEnabled = true;
+				// Sets the information of the expanded image in the state.
+				state.currentImage = state.metadata[ ctx.imageId ];
```

Comment:
Instead of copying the object, let's just save the ID in the state and create a derived state `state.currentImage` that returns the correct object to avoid duplicating the information.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/63348/commits/56d94872c41e57fcbb47652359d3acdedf1bbe63.